### PR TITLE
Cleanup and tweaks

### LINKED ISSUE #91: Make `brew bundle cleanup` safer
I wonder if it makes sense to make `brew bundle cleanup` not do anything without either e.g. `--dry-run`, `--global` or `--force`. A coworker was just trying out `brew bundle` and managed to blow away almost all his installed formulae by mistake. I've personally never used `cleanup` without `--global` and I can see why it would be dangerous.

### PR DESCRIPTION
**Pull request by mikemcquaid**
- General cleanups
- dumper: allow `-` to read from `STDIN`
- cleanup: use `--force` instead of `--dry-run`
- cleanup: also uninstall casks.
- Add check command
- Readme: add example of using `system`.
- gitignore: ignore `bin` directory.
- 100% test coverage.

CC @xu-cheng @josh @technicalpickles

Closes #91 and #100.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
@xu-cheng Whoops, yep, I did (although for consistency I might make it write to `STDOUT`). Any other objections or thoughts?

File: lib/bundle/commands/check.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      current_casks = Bundle::CaskDumper.new.casks
+      requested_casks - current_casks
+    end
+
+    def self.formulae_to_install
+      @@dsl ||= Bundle::Dsl.new(Bundle.brewfile)
+      requested_formulae = @@dsl.entries.select { |e| e.type == :brew }.map(&:name)
+      current_formulae = Bundle::BrewDumper.new.formulae.map { |f| f[:full_name] }
+      upgradable_formulae = Bundle::BrewInstaller.upgradable_formulae
+      requested_formulae - current_formulae - upgradable_formulae
```
Comment (xu-cheng) [inline lib/bundle/commands/check.rb]: You may want to turn all of formulae into short name(as opposite to full name).

**Comment by xu-cheng** [comment]
> although for consistency I might make it write to STDOUT

no objection. If you do this, you need bypass file existing check as `/dev/stdout` will always exist.

File: lib/bundle/commands/check.rb (hunk 1 above)
Comment (mikemcquaid) [inline lib/bundle/commands/check.rb]: Any idea what format `upgradable_formulae` or `Dsl` will give you?

Comment (xu-cheng) [inline lib/bundle/commands/check.rb]: `upgradable_formulae` will give you short names. `DSL` can return both decided by users' Brewfile.

Comment (xu-cheng) [inline lib/bundle/commands/check.rb]: Also why do we need `upgradable_formulae` here? If it's needed, shouldn't it be `requested_formulae - (current_formulae - upgradable_formulae)`?

Comment (mikemcquaid) [inline lib/bundle/commands/check.rb]: I'll play around.

**Comment by xu-cheng** [comment]
Another the thing come to my mind, would `-y` or `--yes` be better than `--force` for cleanup?

**Comment by mikemcquaid** [comment]
@xu-cheng I think that implies there's a prompt without. `--force` feels more appropriate given the destructiveness too.

**Comment by xu-cheng** [comment]
:cool: it's just `--force` feels like something you would try after normal invoking failed.

**Comment by mikemcquaid** [comment]
@xu-cheng Addressed feedback. Let me know what you think.

File: lib/bundle/commands/check.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    def self.casks_to_install
+      @@dsl ||= Bundle::Dsl.new(Bundle.brewfile)
+      requested_casks = @@dsl.entries.select { |e| e.type == :cask }.map(&:name)
+      current_casks = Bundle::CaskDumper.new.casks
+      requested_casks - current_casks
+    end
+
+    def self.formulae_to_install
+      @@dsl ||= Bundle::Dsl.new(Bundle.brewfile)
+      requested_formulae = @@dsl.entries.select { |e| e.type == :brew }.map(&:name)
```
Comment (xu-cheng) [inline lib/bundle/commands/check.rb]: You can turn this into short name by `.map { |f| f.split("/").last }`

Comment (mikemcquaid) [inline lib/bundle/commands/check.rb]: Done! Let me know what you think.

File: lib/bundle/dumper.rb [hunk 3]
```diff
@@ -8,14 +8,17 @@ class Dumper
     def initialize
       @brew = BrewDumper.new
       @cask = CaskDumper.new
-      @repo = RepoDumper.new
+      @repo = TapDumper.new
```
Comment (xu-cheng) [inline lib/bundle/dumper.rb]: Small nit `repo` -> `tap`. Everything else are good :+1:

**Pull request merged**